code/src/containers/cart-drawer/CartDrawer.tsx

```diff
@@ -1,54 +1,54 @@
 import KeyboardArrowLeftIcon from "@mui/icons-material/KeyboardArrowLeft";
 
 import CartDrawerItem from "@/containers/cart-drawer/cart-drawer-item/CartDrawerItem";
-import { mockCartItems } from "@/containers/cart-drawer/constants";
 
 import AppBox from "@/components/app-box/AppBox";
 import AppButton from "@/components/app-button/AppButton";
 import AppIconButton from "@/components/app-icon-button/AppIconButton";
+import AppLoader from "@/components/app-loader/AppLoader";
 import AppTypography from "@/components/app-typography/AppTypography";
 
 import { useDrawerContext } from "@/context/drawer/DrawerContext";
-import useSnackbar from "@/hooks/use-snackbar/useSnackbar";
-import { useRemoveFromCartMutation } from "@/store/api/cartApi";
-import { useUserDetailsSelector } from "@/store/slices/userSlice";
+import useCartItems from "@/hooks/use-cart-items/useUserCartItems";
 import { CartItem } from "@/types/cart.types";
 import formatPrice from "@/utils/format-price/formatPrice";
 
 import "@/containers/cart-drawer/CartDrawer.scss";
 
 const CartDrawer = () => {
-  const user = useUserDetailsSelector();
+  const { closeDrawer } = useDrawerContext();
+  const { id, cartItems, cartItemsLoading, error, handleRemoveItem } =
+    useCartItems();
 
-  const [removeItem] = useRemoveFromCartMutation();
+  //@TODO Create interaction with unauthorization user
+  if (!id) {
+    return null;
+  }
 
-  const { closeDrawer } = useDrawerContext();
+  // @TODO Implement Skeleton for loading items
+  if (cartItemsLoading) return <AppLoader />;
 
-  const { openSnackbarWithTimeout } = useSnackbar();
+  if (error) return <AppTypography translationKey="error.label" />;
 
-  const handleRemoveItem = async (product: CartItem) => {
-    try {
-      if (user?.id) {
-        await removeItem({
-          userId: user.id,
-          productId: product.productId
-        }).unwrap();
-      }
-    } catch {
-      openSnackbarWithTimeout({
-        variant: "error",
-        messageTranslationKey: "cart.itemDeletion.fail"
-      });
-    }
-  };
+  const cartItemsList =
+    cartItems?.items.map((item: CartItem) => (
+      <CartDrawerItem
+        key={item.productId}
+        onRemove={handleRemoveItem}
+        {...item}
+      />
+    )) ?? [];
 
-  const cartItems = mockCartItems.map((item) => (
-    <CartDrawerItem
-      key={item.productId}
-      onRemove={handleRemoveItem}
-      {...item}
-    />
-  ));
+  const renderCartItems =
+    cartItemsList.length > 0 ? (
+      cartItemsList
+    ) : (
+      <AppTypography variant="subtitle2" translationKey="cart.emptyItem" />
+    );
```

Comment:
I suggest to rename it to `cartItemsContent` because with name `renderCartItems` seems like function to me